catalog/src/main/kotlin/com/adevinta/spark/catalog/configurator/samples/buttons/ButtonsConfigurator.kt

```diff
@@ -0,0 +1,321 @@
+    description = "Button configuration",
+    sourceUrl = "$SampleSourceUrl/ButtonSamples.kt",
+) {
+    ButtonSample()
+}
+
+@Preview(
+    showBackground = true,
+)
+@Composable
+private fun ButtonSample() {
+    val scrollState = rememberScrollState()
+    Column(
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+        modifier = Modifier.verticalScroll(scrollState),
+    ) {
+        var icon: SparkIcon? by remember { mutableStateOf(null) }
+        var isLoading by remember { mutableStateOf(false) }
+        var isEnabled by remember { mutableStateOf(true) }
+        var iconSide by remember { mutableStateOf(IconSide.START) }
+        var style by remember { mutableStateOf(ButtonStyle.Filled) }
+        var size by remember { mutableStateOf(ButtonSize.Medium) }
+        var intent by remember { mutableStateOf(ButtonIntent.Primary) }
+        var buttonText by remember { mutableStateOf("Filled Button") }
+
+        ConfigedButton(
+            modifier = Modifier.fillMaxWidth(),
+            style = style,
+            buttonText = buttonText,
+            onClick = { isLoading = !isLoading },
+            isLoading = isLoading,
+            size = size,
+            intent = intent,
+            isEnabled = isEnabled,
+            icon = icon,
+            iconSide = iconSide,
+        )
+
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = spacedBy(8.dp),
+        ) {
+            Text(
+                text = "With Icon",
+                modifier = Modifier.weight(1f).padding(bottom = 8.dp),
+                style = SparkTheme.typography.body2.copy(fontWeight = FontWeight.Bold),
+            )
+            FilledIconToggleButton(
+                checked = icon != null,
+                onCheckedChange = {
+                    icon = if (it) SparkIcons.LikeFill else null
+                },
+            ) {
+                Icon(
+                    sparkIcon = SparkIcons.LikeFill,
+                    contentDescription = null,
+                )
+            }
+        }
+        SwitchLabelled(
+            checked = isLoading,
+            onCheckedChange = { isLoading = it },
+        ) {
+            Text(
+                text = "Loading",
+                modifier = Modifier.fillMaxWidth(),
+            )
+        }
+        SwitchLabelled(
+            checked = isEnabled,
+            onCheckedChange = { isEnabled = it },
+        ) {
+            Text(
+                text = "Enabled",
+                modifier = Modifier.fillMaxWidth(),
+            )
+        }
+
+        Column {
+            Text(
+                text = "IconSize",
+                modifier = Modifier.padding(bottom = 8.dp),
+                style = SparkTheme.typography.body2.copy(fontWeight = FontWeight.Bold),
+            )
+            val iconsSides = IconSide.values()
+            val iconsSidesLabel = iconsSides.map { it.name }
+            SegmentedButton(
```

Comment:
Just a thought: Should we rather use one of the spark's component? like drop down menu?